addons/Rakugo/lib/systems/Executer.gd

```diff
@@ -246,29 +244,55 @@ func do_execute_script(parameters:Dictionary):
 		
 			"SET_VARIABLE":
 				var rvar_name = result.get_string("rvar_name")
-				var text = result.get_string("text")
+				var operator = result.get_string("operator")
+				var final_value = null
+
+				var values = []
+				for var_name in line[3]:
+					var var_ = Rakugo.get_variable(var_name)
+
+					if !var_:
+						parameters["error"] = "Executer::do_execute_script::SET_VARIABLE, can not get variable :" + var_name
+						parameters["stop"] = true
+						break
+
+					values.push_back(var_)
+
+				final_value = line[2].execute(values)
 				
-				var value
+				if line[2].has_execute_failed():
+					parameters["error"] = "Executer::do_execute_script::SET_VARIABLE, failed to execute expression : " + result.get_string("expression")
+					parameters["stop"] = true
+					break
 				
-				if !rvar_name.is_empty():
-					value = Rakugo.get_variable(rvar_name)
+				if operator != "=":
+					if Rakugo.has_variable(rvar_name):
+						var org_value = Rakugo.get_variable(rvar_name)
+						
+						match operator:
+							"+=":
+								Rakugo.set_variable(rvar_name, org_value + final_value)
+							
+							"-=":
+								Rakugo.set_variable(rvar_name, org_value - final_value)
+							
+							"*=":
+								Rakugo.set_variable(rvar_name, org_value * final_value)
+							
+							"/=":
+								Rakugo.set_variable(rvar_name, org_value / final_value)
+							
+							"%=":
+								Rakugo.set_variable(rvar_name, org_value % final_value)
 					
-					if !value:
-						parameters["error"] = "Executer::do_execute_script::SET_VARIABLE, can not get variable :" + rvar_name
+					else:
+						parameters["error"] = "Executer::do_execute_script::SET_VARIABLE, variable not found : " + rvar_name
 						parameters["stop"] = true
 						break
-						
-				elif !text.is_empty():
-					value = remove_double_quotes(text)
-				else:
-					value = result.get_string("number")
 
-					if value.is_valid_int():
-						value = int(value)
-					else:
-						value = float(value)
-
-				Rakugo.set_variable(result.get_string("lvar_name"), value)
+				else:
+					Rakugo.set_variable(rvar_name, final_value)
+				
```

Comment:
You do not need if/else here, just a if.

You can do
```gd
if operator != "=":
  #...
  match:
    "+=":
      final_value += org_value
#endif
Rakugo.set_variable(rvar_name, final_value)
```

Reply:
You are right I will remove else

Reply:
Yes, it should work I will try that.

Reply:
Is done along a go.